code/__HELPERS/icon_smoothing.dm

```diff
@@ -265,9 +265,7 @@
 
 	var/area/target_area = get_area(target_turf)
 	var/area/source_area = get_area(source)
-	if(source_area.canSmoothWithAreas && !is_type_in_typecache(target_area, source_area.canSmoothWithAreas))
-		return null
-	if(target_area.canSmoothWithAreas && !is_type_in_typecache(source_area, target_area.canSmoothWithAreas))
+	if((source_area.area_limited_icon_smoothing || target_area.area_limited_icon_smoothing) && !istype(source_area, target_area.type) && !istype(target_area, source_area.type))
```

Comment:
```suggestion
	if((source_area.area_limited_icon_smoothing || target_area.area_limited_icon_smoothing) && !istype(source_area, target_area.type))
```
`!istype(source_area, target_area.type)` same as `!istype(target_area, source_area.type)`

Reply:
It's not the same, because `istype` returns true for sub-types.
Consider having `/area/shuttle/battleship` and `/area/shuttle/battleship/cockpit`
If we just compare for equality of types, the cockpit won't smooth with the rest of the shuttle, as it happens in the current system.
If we check `istype` for both sides, it will work as it currently does.

Reply:
Ah, great, i forgot about this and this must work nice.